src/library_stack_trace.js

```diff
@@ -1,31 +1,47 @@
 var LibraryStackTrace = {
 
-#if ASSERTIONS && MINIMAL_RUNTIME
-  $demangle__deps: ['$warnOnce'],
+  _demangle_failed: 0,
+
+  $demangle__deps: [
+#if MINIMAL_RUNTIME
+  '$stackSave', '$stackAlloc', '$stackRestore'
+#if ASSERTIONS
+    , '$warnOnce'
+#endif
+#endif
+#if DEMANGLE_SUPPORT
+#if MINIMAL_RUNTIME
+    ,
+#endif
+    '_demangle_failed'
 #endif
+  ],
   $demangle: function(func) {
 #if DEMANGLE_SUPPORT
+    // If demangle has failed before, stop demangling any further function names
+    // This avoids an infinite recursion with malloc()->abort()->stackTrace()->demangle()->malloc()->...
+    if (__demangle_failed) return func;
```

Comment:
I think the `__demangle_failed` global can be simplified to `demangle.failed`, that is, a property on the function itself. Then there is no need to declare it, depend on it, etc. (I think of `func.property` as the same as a `static` variable inside a function in C.)

Reply:
check

Reply:
(in general I try to stay away from `demangle.failed` style of code, since that generates larger code size than `__demangled_failed`, -2 chars per reference if the variable is referenced at least twice.. but in this case definitely doesn't matter.)